.github/workflows/qa.yaml

```diff
@@ -260,3 +262,29 @@ jobs:
         with:
           directory: ${{ matrix.subproject }}/coverage/codecov
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  codecoverage:
+    name: "Collect and upload code coverage"
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
```

Comment:
I think we should only report coverage if none of the previous jobs fail.

Quite often CI fails because we hit some API rate limits when downloading protoc. Can also happen due flaky tests. Under those conditions we'd report incomplete coverage. That's even worse when it happens after a merge, as the tip of the main branch will have misleading coverage and we're not immediately aware of it.